snappy/hwaccess.go

```diff
@@ -53,6 +53,18 @@ func validDevice(device string) bool {
 	return false
 }
 
+func validDeviceForUdev(device string) bool {
+	inValidPrefixes := []string{"/sys/devices", "/sys/class/gpio"}
```

Comment:
A couple more details here: the entries in the list should have a slash suffix to ensure the logic has the intended meaning. Also, are we certain the path reaches this code path clean (no double slashes, etc)?